ckan/freeze-requirements.sh

```diff
@@ -11,17 +11,22 @@ set -e
 USER_ID=$1
 GROUP_ID=$2
 
+# Make sure poetry is installed
+pip3 install poetry
+
 cd /requirements
 
 echo "Running poetry lock ..."
 poetry lock -vvv
 echo "Running poetry export ..."
 poetry export -vvv --format requirements.txt --output requirements.txt --without-hashes
 
-# sadness -- pyz3950 is so old that poetry can't make sense of it's setup.py, so
-# we have to add the requirement manually
-
-echo "-e git+https://github.com/asl2/PyZ3950.git@c2282c73182cef2beca0f65b1eb7699c9b24512e#egg=PyZ3950" >> requirements.txt
+# Sadness: poetry does not keep the "editable" mode that datagovcatalog is installed in.
+# Replace the line with the necessary editable flag
+sed -i 's/ckanext-datagovcatalog \@ git+https\:\/\/github.com\/GSA\/ckanext-datagovcatalog.git\@main/-e git+https:\/\/github.com\/GSA\/ckanext-datagovcatalog.git@main#egg=ckanext-datagovcatalog/g' requirements.txt
+# TODO: add the rest of the editably installed extensions
+# TODO: poetry does not pin to a github commit if given a branch;
+#       need to figure out how to pin appropriately
```

Comment:
@jbrown-xentity Why would pinning to a branch translate to pinning to a commit.  If the branch is updated, isn't the normal functionality to use the most recent commit on that branch?

Reply:
What we don't want to happen is for something to be able to build and/or deploy without being tested. Pinning to a commit solves that problem. We already have major changes that need to be made to datajson and datagovtheme that were not caught in their own tests...